### ISSUE TITLE
Color-contrast false positive due to incorrect element stack

### ISSUE THREAD

**Issue by straker** [dequelabs/axe-core#3666]
### Product

axe-core

### Product Version

4.4.3

### Latest Version

- [X] I have tested the issue with the latest version of the product

### Issue Description

Color contrast shows a false positive for the following scan:

https://axe.deque.com/issues/8e8a526a-bdfb-490a-8bb5-2df499fdb0d5

What's happening is that the element stack code thinks the transparent white overlay on top of the background image of the modal is higher in the stack than it should be (higher than the black background the link is on). Thus it's including the transparent color in the calculation of the background when it shouldn't be (resulting in a lighter color, which fails contrast).

<img width="598" alt="image" src="https://user-images.githubusercontent.com/2433219/191609014-f282904a-f3b4-448f-9fa6-21cf2a5d67e3.png">

#### How to Reproduce

1. In incognito, go to [sijohome.com](http://sijohome.com/)
2. Wait until you receive the first modal, and enter the email `jmrp_test@gmail.com`, then click the "Activate 15% Off" button. It would say that it’s already in use.
3. Dismiss the modal by using the top right x icon
4. The modal in question should show after

#### Additional context

Result of `elementsFromPoint`

```js
rect = $0.getBoundingClientRect()
document.elementsFromPoint(rect.left + rect.width / 2, rect.top + rect.height / 2)

[
  a.css-t1g8t4.e1krutpx0
  div  // <----- Element with the black background
  div#legal.css-3titvn.e1uf35u40
  div#fieldCaptureHeaderBlock.css-1yobj7u.enlhr370
  div#content.css-1iphsiv.e16diobn6
  div.css-yi2lip.emervj1  // <----- Element with the transparent white background
  div#contentframe.css-qbihho.e16diobn2
  div#page1.css-4ss91i.e16diobn0
  div.css-1g4yje1.emervj0
  div#overlayContainer.css-15rn7nd.e1sjo6yw0
  main
  div#dialogContainer
  div#app
  body
  html
]
```

Result of `getElementStack`

```js
axe.setup()
axe.commons.dom.getElementStack($0)

[
  a.css-t1g8t4.e1krutpx0
  div.css-yi2lip.emervj1  // <----- Element with the transparent white background
  div  // <----- Element with the black background
  div#legal.css-3titvn.e1uf35u40
  div#fieldCaptureHeaderBlock.css-1yobj7u.enlhr370
  div#content.css-1iphsiv.e16diobn6
  div#contentframe.css-qbihho.e16diobn2
  div#page1.css-4ss91i.e16diobn0
  div.css-1g4yje1.emervj0
  div#overlayContainer.css-15rn7nd.e1sjo6yw0
  main
  div#dialogContainer
  div#app
  body
  html
]
```

**Comment by straker** [comment]
Going to close this as the modal has been changed and no longer has the same structure, so can no longer test it to figure out what was causing the issue to begin with.
